pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -0,0 +1,134 @@
+
+# Limitations to `<amp-script>`
+
+To keep experiences on AMP pages consistent, limitations exist on `<amp-script>` to guarantee a fast loading time and smooth UI. 
+
+## Initialization
+
+JavaScript inside the Web Worker allows minimal change to the DOM on load. Changes allowed during this phase are:
+
+
+*   Registering event handlers.
+*   Splitting a TextNode into multiple TextNodes to allow for styling liberties, assuming the actual text doesn't change.
+
+The DOM inside `<amp-script>` tags should be almost identical before and after initialization. 
+
+
+## Interaction
+
+To ensure a responsive UX, DOM manipulations in response to a user interaction must happen in **less than one second**.  If a script mutates the DOM in response to a user interactions after one second, the `amp-script` component will terminate the Web Worker. A terminated `<amp-script>` component will not run again.
+
+
+## Script Size 
+
+AMP enforces a limit of 150 kilobytes of custom JavaScript on each page. This limit is shared  between every `<amp-script>` component on that page. If using a library, it must be imported to each individual `<amp-script>` element. 
+
+## Scope
+
+Any DOM elements the custom JavaScript files wishes to interact with must be wrapped inside the `<amp-script>` component tags, this includes other AMP components. The `<amp-script>` component considers `document.body` to be the `amp-script` element and not the document's `<body>` element.
+
+If you were to call `document.body.appendChild(document.createElement('span'))` within the `<amp-script>`file in the following document: 
+
+
+```html
+<body>  
+  <p>Hello!</p>
+  <div>
+    <amp-script layout="container" src="customjs.js">
+    </amp-script>
+  </div>
+</body>
+```
+
+
+It will result in this:
+
+
+```html
+<body>  
+  <p>Hello!</p>
+  <div>
+    <amp-script layout="container" src="customjs.js">
+      <span></span>
+    </amp-script>
+  </div>
+</body>
+```
+
+
+
+## Event Triggers 
+
+JavaScript inserted using `<amp-script>` can listen for the following events:
+
+*   Change
+*   Click
+*   Focus
+*   Blur 
+
+
+## Latency Concerns
+
+Web Workers and the main thread communicate through post [message](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). This can cause a bad user experience with latency, such as use of the `[<canvas>` tag](https://developer.mozilla.org/en-US/docs/Web/API/Canvas_API/Tutorial/Basic_usage). All canvas draw functions would need to be communicated through `postMessage()` to the actual DOM, unintentionally slowing down the page. However, `<amp-script>` supports the use of [basic SVG](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Basic_Shapes). 
+
+TIP: View the[ allowed listed API's here](https://github.com/ampproject/worker-dom/blob/5ef03379c154034545b55f29a2e249f035ee4c8f/src/worker-thread/index.safe.ts#L50-L127).
```

Comment:
This link needs to be updated: https://github.com/ampproject/worker-dom/blob/master/web_compat_table.md